### PR TITLE
drgn: install build dependencies using info from control file

### PR DESCRIPTION
**Pull request by sdimitro**
depends on: https://github.com/delphix/drgn/pull/3

Testing:
http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/linux-pkg-build/job/master/job/userland/job/pre-push/356/

userland-job without the pkg-changes:
http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/linux-pkg-build/job/master/job/userland/job/pre-push/357/

### CODE REVIEW

**Comment by sdimitro** [comment]
cc: @pzakha @prakashsurya

File: packages/drgn/config.sh [hunk 1]
```diff
@@ -23,24 +23,17 @@ UPSTREAM_GIT_URL="https://github.com/osandov/drgn.git"
...
-		zlib1g-dev
+	#
+	# Strictly speaking libkdumpfile is not a hard prerequisite for
+	# drgn itself, but it is a hard requirement in our use-case as
+	# we do want to use drgn for kdump-compressed crash dumps.
+	#
+        if ! dpkg-query --show libkdumpfile >/dev/null 2>&1; then
+                echo_bold "libkdumpfile not installed. Building package 'libkdumpfile' first."
+                logmust "$TOP/buildpkg.sh" libkdumpfile
+        fi
```
Comment (pzakha) [inline packages/drgn/config.sh]: Do we need `libkdumpfile` to be installed on the build machine for properly building `drgn`? If not, then we probably do not need this code.

Comment (sdimitro) [reply packages/drgn/config.sh]: `drgn`'s install scripts check for `libkdumpfile` when compiling the code and if the library exists the some `#define`s are enabled. These `#define`s enable `drgn` to be compiled with support for compressed crash-dumps which we use within Delphix.

Unfortunately this functionality is not something that we can enable at `drgn`'s runtime (at least not yet), thus I wrote this the following way to enforce it as a build dependency.

Does this make sense?

Comment (pzakha) [reply packages/drgn/config.sh]: I see. Yeah, that makes sense, thanks for the explanation. In this case this change looks good.

File: packages/drgn/config.sh [hunk 2]
```diff
@@ -23,24 +23,17 @@ UPSTREAM_GIT_URL="https://github.com/osandov/drgn.git"
...
+
+        logmust install_build_deps_from_control_file
```
[the remark this answers is not in the document]
Comment (sdimitro) [reply packages/drgn/config.sh]: fixed

**Comment by sdimitro** [comment]
userland-build job with both changes succeeded, see link in description.

**Comment by sdimitro** [comment]
running the userland-job without the linux-pkg changes seems to be successful too: http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/linux-pkg-build/job/master/job/userland/job/pre-push/357/

**Pull request merged**